### PR TITLE
ref(settings): Use DropdownMenuControl for sampling rule dropdown

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before:**
Screen Shot 2022-09-15 at 2 57 49 PM

**After:**
Screen Shot 2022-09-15 at 2 55 52 PM
Screen Shot 2022-09-15 at 2 54 33 PM

### CODE REVIEW

**Comment by Jesse-Box** [comment]
Thanks for this @vuluongj20!

**Review by matejminar — APPROVED** [review]
Thanks!

**Comment by priscilawebdev** [comment]
Looks much nicer! 👏

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.spec.tsx [hunk 1]
```diff
@@ -182,9 +182,14 @@ describe('Server-Side Sampling', function () {
     // Assert that project breakdown is there (avoids 'act' warnings)
     expect(await screen.findByText(samplingBreakdownTitle)).toBeInTheDocument();
 
-    const deleteButtons = screen.getAllByLabelText('Delete');
-    expect(deleteButtons[0]).not.toHaveAttribute('disabled'); // eslint-disable-line jest-dom/prefer-enabled-disabled
-    expect(deleteButtons[1]).toHaveAttribute('disabled'); // eslint-disable-line jest-dom/prefer-enabled-disabled
+    userEvent.click(screen.getAllByLabelText('Actions')[0]);
+    let deleteButton = screen.getByRole('menuitemradio', {name: 'Delete'});
+    expect(deleteButton).toHaveAttribute('aria-disabled', 'false');
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.spec.tsx:187]:
```suggestion
    expect(screen.getByRole('menuitemradio', {name: 'Delete'})).toHaveAttribute('aria-disabled', 'false');
```

File: static/app/views/settings/project/server-side-sampling/serverSideSampling.spec.tsx [hunk 2]
```diff
@@ -182,9 +182,14 @@ describe('Server-Side Sampling', function () {
...
+
+    userEvent.click(screen.getAllByLabelText('Actions')[0]);
+    userEvent.click(screen.getAllByLabelText('Actions')[1]);
+    deleteButton = screen.getByRole('menuitemradio', {name: 'Delete'});
+    expect(deleteButton).toHaveAttribute('aria-disabled', 'true');
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/serverSideSampling.spec.tsx:192]:
```suggestion
    expect(screen.getByRole('menuitemradio', {name: 'Delete'})).toHaveAttribute('aria-disabled', 'true');
```

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**